pandas/tests/series/apply/test_series_transform.py

```diff
@@ -1,59 +1,165 @@
+        expected.columns = names
+        result = string_series.transform(ops)
         tm.assert_frame_equal(result, expected)
 
 
-def test_transform_and_agg_error(string_series):
+def test_transform_dict(string_series):
+    # GH 35964
+    with np.errstate(all="ignore"):
+        expected = concat([np.sqrt(string_series), np.abs(string_series)], axis=1)
+    expected.columns = ["foo", "bar"]
+    result = string_series.transform({"foo": np.sqrt, "bar": np.abs})
+    tm.assert_frame_equal(result, expected)
+
+
+def test_transform_udf(axis, string_series):
+    # GH 35964
+    # via apply
+    def func(x):
+        if isinstance(x, Series):
+            raise ValueError
+        return x + 1
+
+    result = string_series.transform(func)
+    expected = string_series + 1
+    tm.assert_series_equal(result, expected)
+
+    # via map Series -> Series
+    def func(x):
+        if not isinstance(x, Series):
+            raise ValueError
+        return x + 1
+
+    result = string_series.transform(func)
+    expected = string_series + 1
+    tm.assert_series_equal(result, expected)
+
+
+def test_transform_wont_agg(string_series):
+    # GH 35964
     # we are trying to transform with an aggregator
-    msg = "transforms cannot produce aggregated results"
+    msg = "Function did not transform"
     with pytest.raises(ValueError, match=msg):
         string_series.transform(["min", "max"])
 
-    msg = "cannot combine transform and aggregation operations"
+    msg = "Function did not transform"
     with pytest.raises(ValueError, match=msg):
         with np.errstate(all="ignore"):
             string_series.transform(["sqrt", "max"])
 
 
 def test_transform_none_to_type():
     # GH34377
-    df = pd.DataFrame({"a": [None]})
-
-    msg = "DataFrame constructor called with incompatible data and dtype"
-    with pytest.raises(TypeError, match=msg):
+    df = DataFrame({"a": [None]})
+    msg = "Transform function failed"
+    with pytest.raises(ValueError, match=msg):
         df.transform({"a": int})
+
+
+def test_transform_reducer_raises(all_reductions):
+    # GH 35964
+    op = all_reductions
+    s = Series([1, 2, 3])
+    msg = "Function did not transform"
+    with pytest.raises(ValueError, match=msg):
+        s.transform(op)
+    with pytest.raises(ValueError, match=msg):
+        s.transform([op])
+    with pytest.raises(ValueError, match=msg):
+        s.transform({"A": op})
+    with pytest.raises(ValueError, match=msg):
+        s.transform({"A": [op]})
+
+
+# mypy doesn't allow adding lists of different types
+# https://github.com/python/mypy/issues/5492
+@pytest.mark.parametrize("op", [*transformation_kernels, lambda x: x + 1])
+def test_transform_bad_dtype(op):
+    # GH 35964
+    s = Series(3 * [object])  # Series that will fail on most transforms
+    if op in ("backfill", "shift", "pad", "bfill", "ffill"):
+        pytest.xfail("Transform function works on any datatype")
```

Comment:
is the issue here that the behavior is wrong or that the test isn't applicable to these cases?

Reply:
Behavior is correct, the test is not applicable. This test is about behavior when transform fails and these ops are expected to never fail.